include/iarc7_fc_comms/CommonFcComms.hpp

```diff
@@ -94,90 +113,165 @@ CommonFcComms<T>& CommonFcComms<T>::getInstance()
+    if (status != FcCommsReturns::kReturnOk) {
+        ROS_ERROR("Failed to retrieve flight controller status");
+        return status;
+    }
     fc.armed = temp_armed;
     fc.auto_pilot = temp_auto_pilot;
     fc.failsafe = temp_failsafe;
-    
-    std_msgs::Float32 battery;
-    #pragma GCC warning "TODO handle failure"
-    flightControlImpl_.getBattery(battery.data);
-
     ROS_DEBUG("Autopilot_enabled: %d", fc.auto_pilot);
     ROS_DEBUG("Armed: %d", fc.armed);
     status_publisher.publish(fc);
+
+    std_msgs::Float32 battery;
+    status = flightControlImpl_.getBattery(battery.data);
+    if (status != FcCommsReturns::kReturnOk) {
+        ROS_ERROR("Failed to retrieve flight controller battery info");
+        return status;
+    }
     ROS_DEBUG("Battery level: %f", battery.data);
     battery_publisher.publish(battery);
 
-
     double attitude[3];
-    flightControlImpl_.getAttitude(attitude);
+    status = flightControlImpl_.getAttitude(attitude);
+    if (status != FcCommsReturns::kReturnOk) {
+        ROS_ERROR("Failed to retrieve attitude from flight controller");
+        return status;
+    }
     ROS_DEBUG("Attitude: %f %f %f", attitude[0], attitude[1], attitude[2]);
     sendOrientationTransform(attitude);
+
+    return FcCommsReturns::kReturnOk;
+}
+
+// Attempt to reconnect, blocking until successful
+template<class T>
+void CommonFcComms<T>::reconnect(const ros::Duration& timeout) {
+    FcCommsReturns status;
+    ros::Time start_time = ros::Time::now();
+    do {
+        status = flightControlImpl_.disconnect();
+    } while (status != FcCommsReturns::kReturnOk
+          && (timeout == ros::Duration(0) || ros::Time::now() - start_time < timeout));
+
+    do {
+        status = flightControlImpl_.connect();
+    } while (status != FcCommsReturns::kReturnOk
+          && (timeout == ros::Duration(0) || ros::Time::now() - start_time < timeout));
 }
 
 // Update the sensors on the flight controller
 template<class T>
 void CommonFcComms<T>::updateSensors(const ros::TimerEvent&)
 {
-
     ros::Time times = ros::Time::now();
 
+    FcCommsReturns status;
 
     // Do different things based on the current connection status.
     switch(flightControlImpl_.getConnectionStatus())
     {
         case FcCommsStatus::kDisconnected:
             ROS_WARN("FC_Comms disconnected");
-            // We don't care about the return value we just reconnect.
-            (void)flightControlImpl_.connect();
+            reconnect();
             break;
-        
+
         case FcCommsStatus::kConnected:
-            #pragma GCC warning "TODO handle failure"
-            flightControlImpl_.handleComms();
-            publishTopics();
+            status = flightControlImpl_.handleComms();
+            while (status != FcCommsReturns::kReturnOk) {
+                reconnect();
+                status = flightControlImpl_.handleComms();
+            }
+
+            while (have_new_direction_command_message_) {
```

Comment:
No because Msp and Common don't handle the fact that some messages overlap. Some things could be sent twice. I'd rather leave it as is and make a trello card to do it right.

Upping it to 20hz might work but you won't be able to guarantee it and it wouldn't be doing it right.